galsim/wfirst/wfirst_psfs.py

```diff
@@ -0,0 +1,460 @@
+                                   interpolating the chromatic object, if `n_waves` is not None.  If
+                                   None, then it uses the blue and red limits of all imaging
+                                   passbands to determine the most inclusive wavelength range
+                                   possible.  But this keyword can be used to reduce the range of
+                                   wavelengths if only one passband (or a subset of passbands) is to
+                                   be used for making the images.
+                                   [default: None]
+    @param    verbose              Should the routine provide information about its progress in
+                                   loading the per-SCA PSFs?  This can be valuable in tracking
+                                   progress, since the full calculations including the real WFIRST
+                                   pupil plane can be quite expensive.  [default: True]
+    @param    achromatic           An option to get an achromatic PSF for a single wavelength, for
+                                   users who do not care about chromaticity of the PSF.  If False,
+                                   then the fully chromatic PSF is returned.  Alternatively the user
+                                   should supply either (a) a wavelength in nanometers, and they
+                                   will get a list of achromatic OpticalPSF objects for that
+                                   wavelength, or (b) a bandpass object, in which case they will get
+                                   a list of achromatic OpticalPSF objects defined at the effective
+                                   wavelength of that bandpass.
+                                   [default: False]
+    @returns  A list of ChromaticOpticalPSF objects, one for each SCA.  The SCAs in WFIRST are
+              1-indexed, and the list is indexed according go the SCA, meaning that the 0th object
+              in the list is always None, the 1st is the PSF for SCA 1, and so on.
+    """
+    # Check which SCAs are to be done.  Default is all.
+    # SCAs are 1-indexed, so we make a list that is one longer than needed, with the 0th element
+    # being None.
+    all_SCAs = np.arange(galsim.wfirst.n_sca + 1)
+    # Later we will use the list of selected SCAs to decide which ones we're actually going to do
+    # the calculations for.  For now, just check for invalid numbers.
+    if SCAs is not None:
+        if hasattr(SCAs, '__iter__'):
+            if min(SCAs) < 0 or max(SCAs) > galsim.wfirst.n_sca:
+                raise ValueError(
+                    "Invalid SCA!  Indices must be positive and <=%d."%galsim.wfirst.n_sca)
+            if min(SCAs) == 0:
+                import warnings
+                warnings.warn("SCA 0 requested, but SCAs are 1-indexed.  Ignoring 0.")
+        else:
+            SCAs = [SCAs]
```

Comment:
Should probably make sure this is a valid value too.  Easiest thing would be to do this before the min/max tests.  Also, it would be simpler later if you do `if SCAs is None: SCAs = all_SCAs`.  Then you can just loop over `SCAa` below without so many checks.